runtime/src/bank.rs

```diff
@@ -9334,13 +9359,25 @@ mod tests {
         assert_eq!(bank.get_account_modified_slot(&program_id).unwrap().1, slot);
 
         let mut bank = Arc::new(new_from_parent(&bank));
+        // When replacing native_program, name must change to disambiguate from repeated
+        // invocations.
```

Comment:
I think we should put a comment about this in builtins.rs, since it's critical to know when constructing the name

Reply:
I'll sneak that into #13394